src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/controllers/v1/person/OffencesController.kt

```diff
@@ -30,8 +29,8 @@ class OffencesController(
     val pncId = encodedPncId.decodeUrlCharacters()
     val response = getOffencesForPersonService.execute(pncId)
 
-    if (response.hasErrorCausedBy(UpstreamApiError.Type.ENTITY_NOT_FOUND, causedBy = UpstreamApi.NOMIS)) {
-      throw EntityNotFoundException("Could not find person with id: $pncId")
+    if (response.hasError(UpstreamApiError.Type.ENTITY_NOT_FOUND)) {
+      throw EntityNotFoundException("Could not find offences for person with id: $pncId")
```

Comment:
I think the error message we had befoar was correct. If an upstream API returns 404, it means that the person with that ID couldn't be found i.e. they don't exist. The message above suggests to me that a person with the ID I gave was found but they had no offences. But we've said if a person is found and has no offences then we 200 and return an empty list.

Reply:
Cool yeah makes sense, updated.